app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryUiRenderer.kt

```diff
@@ -15,6 +19,15 @@ class BloodSugarEntryUiRenderer(
     openAsValueChangedCallback.pass(model.openAs) { setupUi(it) }
     manageProgress(model)
     manageBloodSugarUnitPreferenceButtonText(model)
+    manageBloodSugarUnitPreferenceButtonVisibility(model)
+  }
+
+  private fun manageBloodSugarUnitPreferenceButtonVisibility(model: BloodSugarEntryModel) {
+    when (model.openAs.measurementType) {
+      Random, PostPrandial, Fasting -> ui.showBloodSugarUnitPreferenceButton()
+      HbA1c -> ui.hideBloodSugarUnitPreferenceButton()
+      else -> ui.showBloodSugarUnitPreferenceButton()
```

Comment:
I think we should hide the blood sugar unit preference button for unknown measurement types to avoid any unnecessary conversions. Please check with Vinay once.

Reply:
Pushing the updated changes now. Will change this when Vinay replied in the next PR if required.

Reply:
Let's just hide this, for now, we will continue saving them as `mg` anyway.